polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -4075,3 +4114,59 @@ fn compute_delayed_approval_sending_tick(
 	metrics.on_delayed_approval(sign_no_later_than.checked_sub(tick_now).unwrap_or_default());
 	sign_no_later_than
 }
+
+// collect all the approvals required to approve the
+// candidate, ignoring any other approval that belongs
+// to not required tranches
+fn collect_useful_approvals<Sender>(
+	sender: &mut Sender,
+	status: &ApprovalStatus,
+	block_hash: Hash,
+	candidate_entry: &CandidateEntry,
+)
+where
+	Sender: SubsystemSender<ConsensusStatisticsCollectorMessage>
+{
+	let candidate_hash = candidate_entry.candidate.hash();
+	let candidate_approvals = candidate_entry.approvals();
+
+	let approval_entry = candidate_entry
+		.approval_entry(&block_hash)
+		.expect("Approval entry just fetched; qed");
```

Comment:
Being paranoid here, but this is too far away from `approval_entry` previous call, so let's use if let Some() here and an warn on error path.